zbus/tests/e2e.rs

```diff
@@ -1118,3 +1118,45 @@ async fn iface_and_proxy_(#[allow(unused)] p2p: bool) {
     );
     debug!("Bus confirmed that all names were definitely released.");
 }
+
+struct TimerTest {}
```

Comment:
Just to remind of my exact wording:

> it's probably best to add a method to the e2e test (in zbus/tests/e2e.rs) that tests this, instead of a completely separate test.

You're adding a completely separate test still.

Reply:
I tried my best there, still probably not exactly how you want it. I tried to add more code to `my_iface_test` function but that didn't work well for `p2p` case, so I still had to put a real test code into a separate function. Anyway I think that `my_iface_test` there is already extremely overload and is hard to read.

Reply:
> I tried my best there

And thanks for that. I didn't doubt it. I just feel like you sometimes miss my comments.

> still probably not exactly how you want it.

I want like to believe that it's less about what I want and more about the best solution. If/when you disagree with me on that, please do say so and we can try to come up with the best solution together.

I realize that the interface here is quite bloated already and I would love contributions on refactoring it. Simply adding a new test case for each feature, is not a solution either as that would add to the compilation and test run time, and add duplication.